R/queue.R

```diff
@@ -0,0 +1,78 @@
+#' Object for managing running jobs on the redis queue
+#'
+#' @keywords internal
+Queue <- R6::R6Class("Queue", #nolint
+  cloneable = FALSE,
+  public = list(
+    #' @field root Orderly root
+    root = NULL,
+    #' @field config Orderly config
+    config = NULL,
+
+    #' @description
+    #' Create object, read configuration and setup Redis connection.
+    #'
+    #' @param root Orderly root.
+    #' @param queue_id ID of an existing queue to connect to, creates a new one
+    #'   if NULL (default NULL)
+    #' @param workers Number of workers to spawn (default 1)
+    #' @param cleanup_on_exit If TRUE workers are killed on exit (defaults
+    #'   to TRUE if workers is > 0)
+    initialize = function(root, queue_id = NULL, workers = 1,
+                          cleanup_on_exit = workers > 0) {
+      private$cleanup_on_exit <- cleanup_on_exit
+      self$root <- root
+      self$config <- orderly2::orderly_config(self$root)
+      if (!runner_has_git(self$root)) {
+        cli::cli_abort(paste("Not starting server as orderly",
+                             "root is not version controlled."))
+      }
+
+      # Connecting to Redis
+      private$con <- redux::hiredis()
+
+      # Create queue
+      private$queue_id <- queue_id %||% orderly_queue_id()
+      private$queue <- rrq::rrq_controller$new(private$queue_id,
+                                               private$con)
```

Comment:
```suggestion
      private$queue <- rrq::rrq_controller2(private$queue_id)
```

Uses the new form.  No point passing `con` if you're going to default construct it either.  You might want to call this `controller` not `queue` in the object though as in hintr reading `queue$queue` is always quite confusing

Reply:
done, have named it controller and removed `queue_id` as private variable too since controller exposes it anyway